samples/streaming/provider/src/provider_impl.rs

```diff
@@ -0,0 +1,213 @@
+                continue;
+            }
+            images.push(entry.file_name().into_string().unwrap());
+        }
+
+        if images.is_empty() {
+            panic!("No images");
+        }
+
+        Ok(images)
+    }
+}
+
+#[tonic::async_trait]
+impl DigitalTwinProvider for ProviderImpl {
+    type StreamStream = Pin<Box<dyn Stream<Item = Result<StreamResponse, Status>> + Send>>;
+
+    /// Subscribe implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Subscribe request.
+    async fn subscribe(
+        &self,
+        request: Request<SubscribeRequest>,
+    ) -> Result<Response<SubscribeResponse>, Status> {
+        warn!("Got a subscribe request: {request:?}");
+
+        Err(Status::unimplemented("subscribe has not been implemented"))
+    }
+
+    /// Unsubscribe implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Unsubscribe request.
+    async fn unsubscribe(
+        &self,
+        request: Request<UnsubscribeRequest>,
+    ) -> Result<Response<UnsubscribeResponse>, Status> {
+        warn!("Got an unsubscribe request: {request:?}");
+
+        Err(Status::unimplemented("unsubscribe has not been implemented"))
+    }
+
+    /// Get implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Get request.
+    async fn get(&self, request: Request<GetRequest>) -> Result<Response<GetResponse>, Status> {
+        warn!("Got a get request: {request:?}");
+
+        Err(Status::unimplemented("get has not been implemented"))
+    }
+
+    /// Set implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Set request.
+    async fn set(&self, request: Request<SetRequest>) -> Result<Response<SetResponse>, Status> {
+        warn!("Got a set request: {request:?}");
+
+        Err(Status::unimplemented("set has not been implemented"))
+    }
+
+    /// Invoke implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Invoke request.
+    async fn invoke(
+        &self,
+        request: Request<InvokeRequest>,
+    ) -> Result<Response<InvokeResponse>, Status> {
+        warn!("Got an invoke request: {request:?}");
+
+        Err(Status::unimplemented("get has not been implemented"))
+    }
+
+    /// Stream implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Stream request.
+    async fn stream(
+        &self,
+        _request: Request<StreamRequest>,
+    ) -> Result<Response<Self::StreamStream>, Status> {
+        let image_filenames = self
+            .get_filenames_from_image_directory()
+            .map_err(|err| Status::internal(format!("Get filenames failed due to: {err}")))?;
+
+        let image_file_iterator = ImageFileIterator::new(&self.image_directory, image_filenames);
+
+        let mut stream =
+            Box::pin(tokio_stream::iter(image_file_iterator).throttle(Duration::from_secs(5)));
+
+        // The spawn and channel are required if you want to handle disconnect functionality.
+        let (tx, rx) = mpsc::channel(128);
```

Comment:
Please assign the buffer size to a constant.

Reply:
I don't agree that a constant is needed here, It is never re-used.